test/services/katello/host_trace_manager.rb

```diff
@@ -0,0 +1,64 @@
+require 'katello_test_helper'
+
+unless Katello.with_remote_execution?
+  class JobInvocationComposer
+  end
+end
+
+module Katello
+  class HostTraceManagerTest < ActiveSupport::TestCase
+    def setup
+      @host1 = FactoryBot.create(:host)
+      @host2 = FactoryBot.create(:host)
+    end
+
+    def test_resolve_group_by_helper_one_invocation
+      trace_one = Katello::HostTracer.create(host_id: @host1.id, application: 'rsyslog', app_type: 'daemon', helper: 'systemctl restart rsyslog')
+      trace_two = Katello::HostTracer.create(host_id: @host2.id, application: 'rsyslog', app_type: 'daemon', helper: 'systemctl restart rsyslog')
+      helper = {:helper => trace_two.helper}
+      traces = [trace_one, trace_two]
+      job_invocation = {"description" => "Restart Services", "id" => 1, "job_category" => "Katello"}
+
+      JobInvocationComposer.expects(:for_feature).with(:katello_service_restart, [@host1.id, @host2.id], helper).returns(mock(trigger: true, job_invocation: job_invocation))
+
+      Katello::HostTraceManager.resolve_traces(traces)
```

Comment:
It would be good to ensure that this works with an AR relation as input rather than just an array so you could do this: Katello::HostTracer.where(id: trace.id). Mentioning this since that's how we use this class in the controllers

Reply:
Makes sense, changed to reflect the way we resolve the traces instead of an array in the test and use the AR relation for the resolve.